packages/messaging/android/src/main/java/io/invertase/firebase/messaging/ReactNativeFirebaseMessagingModule.java

```diff
@@ -55,23 +56,29 @@ public void getInitialNotification(Promise promise) {
       initialNotification = null;
       return;
     } else {
-      Intent intent = getCurrentActivity().getIntent();
+      Activity activity = getCurrentActivity();
 
-      if (intent != null && intent.getExtras() != null) {
-        // messageId can be either one...
-        String messageId = intent.getExtras().getString("google.message_id");
-        if (messageId == null) messageId = intent.getExtras().getString("message_id");
+      if (activity != null) {
+        Intent intent = activity.getIntent();
 
-        // only handle non-consumed initial notifications
-        if (messageId != null && initialNotificationMap.get(messageId) == null) {
-          RemoteMessage remoteMessage = ReactNativeFirebaseMessagingReceiver.notifications.get(messageId);
+        if (intent != null && intent.getExtras() != null) {
+          // messageId can be either one...
+          String messageId = intent.getExtras().getString("google.message_id");
+          if (messageId == null) messageId = intent.getExtras().getString("message_id");
 
-          if (remoteMessage != null) {
-            promise.resolve(ReactNativeFirebaseMessagingSerializer.remoteMessageToWritableMap(remoteMessage));
-            initialNotificationMap.put(messageId, true);
-            return;
+          // only handle non-consumed initial notifications
+          if (messageId != null && initialNotificationMap.get(messageId) == null) {
+            RemoteMessage remoteMessage = ReactNativeFirebaseMessagingReceiver.notifications.get(messageId);
+
+            if (remoteMessage != null) {
+              promise.resolve(ReactNativeFirebaseMessagingSerializer.remoteMessageToWritableMap(remoteMessage));
+              initialNotificationMap.put(messageId, true);
+              return;
+            }
           }
         }
+      } else {
+        Log.w(TAG, "Attempt to call getInitialNotification failed. The current activity is not ready, try calling the method later in the React lifecycle.");
```

Comment:
Resolve null here? Otherwise promise will hang. Same in other scenarios (e.g null intent) I guess - missing a resolve

Reply:
Ignore this comment,  promise resolve below